exercises/practice/binary-search-tree/.meta/src/BinarySearchTree.example.elm

```diff
@@ -0,0 +1,40 @@
+module BinarySearchTree exposing (BinaryTree(..), makeTree, sort)
+
+
+type BinaryTree
+    = Nil
```

Comment:
Recommend renaming `Nil` --> `Empty` to avoid confusion with Elm not supporting the concept of nil.

Reply:
I think Nil in a custom type is fairly idiomatic in Elm, although I'm not quite sure. Nothing is used within Maybe, and other modules tend to use something else to avoid a name clash as Maybe is so common, and I think Nil is often what people go for.

Reply:
> ...Nothing is used within Maybe, and other modules tend to use something else to avoid a name clash as Maybe is so common, and I think Nil is often what people go for.

I was recommending a custom type 'Empty' for that reason. Either way, I don't think it's a significant point, just a more idiomatic option.

Reply:
I also thought `Nil` was quite idiomatic in Elm, but most tutorials I could find in my extensive 5-minute search reported using `Empty`. Maybe it's more common in Haskell?
Anyway, made the change in [d09f956](https://github.com/exercism/elm/pull/609/commits/d09f956382f5cc6e15b885273b637363fd2db751)